packages/malloy/src/lang/malloy-to-ast.ts

```diff
@@ -223,23 +224,10 @@ export class MalloyToAST
   }
 
   protected getFieldDefs(
-    cxList: ParserRuleContext[],
-    isAgg?: boolean
+    cxList: parse.FieldDefContext[],
+    makeFieldDef: ast.FieldDeclarationConstructor
```

Comment:
Changed the way that field declaration "subtypes" work. Now instead of creating a generic `FieldDeclaration` and then attaching subtype info, we just construct different subclasses of `FieldDeclaration` depending on the place the field declaration occurs. This allows us to do nice typechecking of what kinds of fields are used in different kinds of definitions.